src/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -310,13 +359,6 @@ await _prefillService.PrefillDataModel(
                 throw new InvalidOperationException("Expected exactly one change in initialChanges");
             }
 
-            // Mutates initialChanges and to add DataElement Type = Created
-            await dataMutator.UpdateInstanceData(initialChanges);
-
-            var newDataElement =
-                change.DataElement
-                ?? throw new InvalidOperationException("DataElement not set in dataMutator.UpdateInstanceData");
-
```

Comment:
Moving from two `updateInstanceData` to one, fixed the duplicate creation. I used the one to make it easier to figure out what changed data element was from the post, and what addition might have been performed in dataProcessing. The solution was to just keep a reference to the change and update the change when the data element was available.